rest/blip_client_test.go

```diff
@@ -484,23 +531,85 @@ func createBlipTesterClientOpts(tb testing.TB, rt *RestTester, opts *BlipTesterC
 	btc := BlipTesterClient{
 		BlipTesterClientOpts: *opts,
 		rt:                   rt,
-		docs:                 make(map[string]map[string]*BodyMessagePair),
-		attachments:          make(map[string][]byte),
-		lastReplicatedRev:    make(map[string]string),
 	}
 
 	id, err := uuid.NewRandom()
 	if err != nil {
 		return nil, err
 	}
 
-	if btc.pushReplication, err = newBlipTesterReplication(tb, "push"+id.String(), &btc); err != nil {
+	if btc.pushReplication, err = newBlipTesterReplication(btc.rt.tb, "push"+id.String(), &btc); err != nil {
 		return nil, err
 	}
-	if btc.pullReplication, err = newBlipTesterReplication(tb, "pull"+id.String(), &btc); err != nil {
+	if btc.pullReplication, err = newBlipTesterReplication(btc.rt.tb, "pull"+id.String(), &btc); err != nil {
 		return nil, err
 	}
 
+	btc.CollectionClients = make(map[string]*BlipTesterCollectionClient)
+
+	if len(opts.Collections) != 0 {
+		getCollectionsRequest := blip.NewRequest()
+		getCollectionsRequest.SetProfile(db.MessageGetCollections)
+
+		type CollectionsRequest struct {
```

Comment:
dupe of https://cs.github.com/couchbase/sync_gateway/blob/5efe9e8dcbb3836091c7b93e0d85acccbdb43cad/db/blip_sync_messages.go?q=collectionsrequest#L491